hemco_interface.F90

```diff
@@ -1427,59 +1423,56 @@ subroutine HCO_GC_Run(GC, IMPORT, EXPORT, Clock, RC)
         endif
 
         !-----------------------------------------------------------------------
-        ! Get time properties
+        ! Get time properties. Current time is time at end of current timestep.
+        ! Previous time is time at start of current timestep.
         !-----------------------------------------------------------------------
-        call get_prev_time(prev_day, prev_s)
-        call get_curr_time(now_day, now_s)
-        call get_curr_date(year, month, day, tod)
+        call get_prev_time(ts0_day, ts0_s)
+        call get_prev_date(ts0_year, ts0_month, ts0_day, ts0_tod)
+        call get_curr_time(ts1_day, ts1_s)
+        call get_curr_date(ts1_year, ts1_month, ts1_day, ts1_tod)
 
+        ! Optional debug prints to display CAM times
         !if(masterproc) then
-        !    write(iulog,*) "hco year,month,day,tod", year, month, day, tod
-        !    write(iulog,*) "hco prev_day, prev_s", prev_day, prev_s
-        !    write(iulog,*) "hco now_day, now_s", now_day, now_s
+        !   write(iulog,*) "HEMCO_CESM debug: CAM date at start of timestep: year, month, day, tod: ", ts0_year, ts0_month, ts0_day, ts0_tod
+        !   write(iulog,*) "HEMCO_CESM debug: CAM date at end of timestep: year, month, day, tod: ", ts1_year, ts1_month, ts1_day, ts1_tod
+        !   write(iulog,*) "HEMCO_CESM debug: CAM time at start of timestep: day, s: ", ts0_day, ts0_s
+        !   write(iulog,*) "HEMCO_CESM debug: CAM time at end of timestep: day, s: ", ts1_day, ts1_s
         !endif
-        ! 2005 1 1 1800 | 0 0 | 0 1800
-        ! 2005 1 1 3600 | 0 1800 | 0 3600
-        ! 2005 1 1 5400 | 0 3600 | 0 5400
-        ! ...
-
-        ! Check if we have run HEMCO for this time step already. If yes can exit
-        if(last_HCO_day * 86400.0 + last_HCO_second .ge. now_day * 86400.0 + now_s) then
-            ! The first timestep should not be skipped. A fix is made during initialization (hplin, 6/11/24)
-            if(masterproc) then
-                write(iulog,*) "HEMCO_CESM: HEMCO already ran for this time, check timestep mgr (now day, s; last day, s)", now_day, now_s, last_HCO_day, last_HCO_second
-            endif
 
-            return
+        ! Check if we have run HEMCO for this time step already based on CAM time
+        ! which is time at end of timestep
+        if ( ( ts0_day == ts1_day ) .and. ( ts0_s == ts1_s ) ) then
+           if(masterproc) write(iulog,*) "HEMCO_CESM: CAM current and previous times are the same. Do not run HEMCO yet."
```

Comment:
Can I ask if you could elaborate further on the comment for this `if` or the debug message? I'm assuming from "do not run HEMCO yet" this means that `( ts0_day == ts1_day ) .and. ( ts0_s == ts1_s )` matches some kind of time condition at the beginning of a CAM run...? What is special about this condition that HEMCO shouldn't be ran? Thanks!

Reply:
Looks like I never updated this comment. It should say check that CAM has started timestepping. I removed the check for if HEMCO has already run since it seems unnecessary. Please let me know if you disagree! The new check prevents HEMCO from running until initialization in CAM is complete. There is a timestep 0 that occurs the first run (CONTINUE_RUN=false). When timestep 0 is in progress the current and previous CAM times are the same. Timestepping officially begins when current time becomes end of timestep.

Reply:
I don't quite recall why there was a check for HEMCO has already run - maybe it was in anticipation of a multiple CAM instance functionality, or maybe it was related to timestep 0. I don't object removing it as long as the timestepping behavior is correct. Thanks!